datastore_version_manager/domain/metadata_patcher.py

```
@@ -1,14 +1,47 @@
+                     exclude_regex_paths=[
+                         r"root\['populationDescription'\]",
+                         r"root\['identifierVariables'\]\[\d+\]\['name'\]",
+                         r"root\['identifierVariables'\]\[\d+\]\['description'\]",
+                         r"root\['identifierVariables'\]\[\d+\]\['unitType'\]\['name'\]",
+                         r"root\['identifierVariables'\]\[\d+\]\['unitType'\]\['description'\]",
+                         r"root\['identifierVariables'\]\[\d+\]\['valueDomain'\]\['description'\]",
+                         r"root\['measureVariables'\]\[\d+\]\['name'\]",
+                         r"root\['measureVariables'\]\[\d+\]\['description'\]",
+                         r"root\['measureVariables'\]\[\d+\]\['subjectFields'\]\[\d+\]\['name'\]",
+                         r"root\['measureVariables'\]\[\d+\]\['subjectFields'\]\[\d+\]\['description'\]",
+                         r"root\['measureVariables'\]\[\d+\]\['valueDomain'\]\['codeList'\]\['codeItems'\]\[\d+\]"
+                         r"\['categoryTitle'\]",
+                         r"root\['measureVariables'\]\[\d+\]\['valueDomain'\]\['uriDefinition'\]",
+                         r"root\['measureVariables'\]\[\d+\]\['valueDomain'\]\['sentinelAndMissingValues'\]\[\d+\]"
+                         r"\['categoryTitle'\]",
+                         r"root\['measureVariables'\]\[\d+\]\['valueDomain'\]\['description'\]",
+                         r"root\['measureVariables'\]\[\d+\]\['valueDomain'\]\['measurementUnitDescription'\]",
+                         r"root\['attributeVariables'\]\[\d+\]\['name'\]",
+                         r"root\['attributeVariables'\]\[\d+\]\['description'\]",
+                         r"root\['attributeVariables'\]\[\d+\]\['valueDomain'\]\['description'\]"
+                     ])
+
+    if 'dictionary_item_added' in ddiff:
+        raise Exception(f"Is is not allowed to add new metadata fields: {ddiff['dictionary_item_added']}")
```

Can we raise an explicit exception here? PatchMetadataError or similar

EDIT: and then catch it explicitly in the test below